service-node/backend/src/main/scala/org/enterprisedlt/fabric/service/node/rest/UploadHandler.scala

```diff
@@ -0,0 +1,90 @@
+package org.enterprisedlt.fabric.service.node.rest
+
+import java.nio.file.{Files, Path, Paths, StandardOpenOption}
+
+import javax.servlet.MultipartConfigElement
+import javax.servlet.http.{HttpServletRequest, HttpServletResponse}
+import org.eclipse.jetty.http.MimeTypes
+import org.eclipse.jetty.server.Request
+import org.eclipse.jetty.server.handler.AbstractHandler
+import org.eclipse.jetty.util.IO
+import org.enterprisedlt.fabric.service.node.Util
+import org.slf4j.LoggerFactory
+
+import scala.util.Try
+
+/**
+ * @author Alexey Polubelov
+ */
+
+
+class UploadHandler(
+    multipartConfig: MultipartConfigElement,
+    endpoints: Array[FileUploadEndpoint],
+    createCodec: () => ServerCodec,
+) extends AbstractHandler {
+
+    private val logger = LoggerFactory.getLogger(this.getClass)
+
+    override def handle(target: String, baseRequest: Request, request: HttpServletRequest, response: HttpServletResponse): Unit = {
+        if (request.getMethod == "POST" && endpoints.exists(_.uri == request.getPathInfo)) {
+            RestEndpointContext.set(RestEndpointContext(request, response))
+            request.setAttribute(Request.MULTIPART_CONFIG_ELEMENT, multipartConfig)
+            processRequest(endpoints, request, response)
+            RestEndpointContext.clean()
+            baseRequest.setHandled(true)
+        }
+
+    }
+
+    private def processRequest(
+        endpoints: Array[FileUploadEndpoint],
+        request: HttpServletRequest, response: HttpServletResponse
+    ): Unit = {
+        if (endpoints.exists(_.uri == request.getPathInfo)) {
```

Comment:
you already did this check, consider reuse

Reply:
done.